### PR TITLE
add opencv4 into the dependency list for ROS2

### PR DESCRIPTION
**Pull request by LewisLiuPub**
@gaoethan @ahcorde
I updated code to support OpenCV4 for branch ros2, please have a review.

Signed-off-by: Lewis Liu <wei.zhi.liu@intel.com>

### CODE REVIEW

File: cv_bridge/CMakeLists.txt [hunk 1]
```diff
@@ -37,7 +37,7 @@ else()
 endif()
 
 find_package(sensor_msgs REQUIRED)
-find_package(OpenCV 3 REQUIRED
+find_package(OpenCV 4 REQUIRED
```
Comment (ahcorde) [inline cv_bridge/CMakeLists.txt]: This line force to OpenCV4 , there is no option for version 3.

You should do something like:

```
find_package(OpenCV 4 QUIET opencv_core opencv_core)
if(NOT OpenCV_FOUND)
  find_package(OpenCV 3 REQUIRED opencv_core opencv_core)
endif()
```

**Review by ahcorde — CHANGES_REQUESTED** [review]
Please review the comment regarding OpenCV

File: cv_bridge/CMakeLists.txt (hunk 1 above)
Comment (LewisLiuPub) [reply cv_bridge/CMakeLists.txt]: It really works, I checked it and uploaded one new patch.
@ahcorde

**Review by chapulina — APPROVED** [review]
FWIW this works for me with OpenCV 4 on Focal.

**Comment by ruffsl** [comment]
@mjcarroll @ahcorde did you get a chance to look at this?
I see in https://github.com/ros-perception/vision_opencv/issues/272 you've been discussing about this for foxy release.

**Review by ahcorde — APPROVED** [review]

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**